src/server/jobs/collection-contest-vimeo-upload.ts

```diff
@@ -0,0 +1,198 @@
+                },
+                body: JSON.stringify({
+                  imageId: item.imageId,
+                  sourceUrl: getEdgeUrl(item.imageUrl, {
+                    original: true,
+                  }),
+                  callbackUrl,
+                }),
+              });
+
+              if (!res.ok) {
+                throw new Error(`Failed to upload video ${item.imageId}`);
+              }
+
+              await dbWrite.image.update({
+                where: { id: item.imageId },
+                data: {
+                  metadata: {
+                    ...item.metadata,
+                    vimeoUploadEnqueuedAt: new Date().toISOString(),
+                  },
+                },
+              });
+            } else {
+              // Assume we're uploading from the CRON Job:
+              const userProfile = `${env.NEXT_PUBLIC_BASE_URL}/user/${item.username}`;
+
+              const data: { link: string } = await uploadVimeoVideo({
+                url: item.imageUrl,
+                accessToken: env.VIMEO_ACCESS_TOKEN,
+                title: item.title,
+                description: `
+                  ${sanitize(item.detail, {
+                    allowedTags: [],
+                    allowedAttributes: {},
+                  })}
+
+                  Created by ${item.username}:
+                  ${userProfile}
+
+                  Check out more entries at:
+                  ${env.NEXT_PUBLIC_BASE_URL}/collections/${collection.id}
+                `,
+                size: item.metadata.size as number,
+              });
+
+              const { link } = data;
+              const videoId = link.split('/').pop();
+
+              await dbWrite.image.update({
+                where: { id: item.imageId },
+                data: {
+                  metadata: {
+                    ...item.metadata,
+                    vimeoVideoId: videoId,
+                  },
+                },
+              });
+            }
+          } catch (error) {
+            console.error(`Error uploading video ${item.imageId}: ${(error as Error).message}`);
+            logWebhook({ error, imageId: item.imageId });
+
+            await dbWrite.image.update({
+              where: { id: item.imageId },
+              data: {
+                metadata: {
+                  ...item.metadata,
+                  vimeoUploadAttempt: (item.metadata.vimeoUploadAttempt ?? 0) + 1,
+                },
+              },
+            });
+
+            await dbWrite.collectionItem.updateMany({
+              where: {
+                imageId: item.imageId,
+                collectionId: collection.id,
+              },
+              data: {
+                updatedAt: new Date(),
+              },
+            });
+          }
+        }
+      } catch (error) {
+        console.error(`Error processing collection ${collection.id}: ${(error as Error).message}`);
+        logWebhook({ error });
+      }
+    }
+
+    // await setLastRun();
```

Comment:
Don't we want to set the timestamp when this completed? I guess it would be wise to just do it within the happy path only
```suggestion
    await setLastRun();
```